src/processors/models/processorsManager.js

```diff
@@ -34,12 +34,41 @@ module.exports.getProcessor = async function(processorId) {
     if (processor) {
         return processor;
     } else {
-        const error = new Error(ERROR_MESSAGES.NOT_FOUND);
-        error.statusCode = 404;
+        const error = generateProcessorNotFoundError();
         throw error;
     }
 };
 
 module.exports.deleteProcessor = async function (processorId) {
     return databaseConnector.deleteProcessor(processorId);
 };
+
+module.exports.redownloadJSProcessor = async function(processorId) {
+    const processor = await databaseConnector.getProcessor(processorId);
+    try {
+        if (!processor) {
+            const error = generateProcessorNotFoundError();
```

Comment:
throwing an error inside a `try {...}` block and then catching it right after isn't needed. I would use the methods that are already implemented inside the processorsManager such as `getProcessor` where the error logic is written instead of calling the dbConnected. (Id also implement updateProcessor method in the manager, which will also implement the error logic

Then you can have one block of `try {}` without throwing any errors. The methods inside the try block will be the ones throwing errors and if they do, you will catch it in the catch statement and log `Error occurred trying to re-download processor file`

Reply:
done